provider/openstack/provider.go

```diff
@@ -648,6 +649,11 @@ func (e *Environ) Bootstrap(ctx environs.BootstrapContext, args environs.Bootstr
 func (e *Environ) supportsNeutron() bool {
 	client := e.client()
 	endpointMap := client.EndpointsForRegion(e.cloud.Region)
+	if len(endpointMap) == 0 {
+		// In some corner cases the endpoingMap is empty.  Default to
```

Comment:
It turns out that even though the callers to supportsNeutron() were authenticating the client before calling.  The client used by suportsNeutron() was not alway authenticated.  There is a bigger question of why that should be looked at?  I moved code around so that supportsNeutron() was checking for client authentication for now.